fuzzers/main.py

```diff
@@ -0,0 +1,949 @@
+    print(f' -> picked {res}', file=sys.stderr)
+    return res
+
+
+def pause_exit_spot(pause_evt: threading.Event, resume_evt: threading.Event,
+                    exit_evt: threading.Event) -> bool:
+    """
+    Poll the events, returning True if `exit_evt` was set and pausing as requested by `pause_evt`
+    and `resume_evt`
+
+    Args:
+        pause_evt: the pausing event to poll
+        resume_evt: the resuming event to poll
+        exit_evt: the exiting event to poll
+    """
+
+    if pause_evt.is_set():
+        while not resume_evt.wait(timeout=1):
+            if exit_evt.is_set():
+                return True
+    return exit_evt.is_set()
+
+
+def kill_fuzzers(bucket: gcs.Bucket,
+                 fuzzers: typing.Iterable[FuzzProcess]) -> None:
+    """Kill all the fuzzers from `fuzzers`, uploading their logs to `bucket`
+
+    Args:
+        bucket: the bucket  to submit the logs to
+        fuzzers: the list of fuzzers to kill
+    """
+    for fuzzer in fuzzers:
+        try:
+            fuzzer.signal(signal.SIGTERM)
+        except ProcessLookupError:
+            print(f'Failed looking up process {fuzzer.proc.pid}',
+                    file=sys.stderr)
+    time.sleep(5)
+    for fuzzer in fuzzers:
+        try:
+            fuzzer.signal(signal.SIGKILL)
+        except ProcessLookupError:
+            pass
+    for fuzzer in fuzzers:
+        bucket.blob(f'logs/{fuzzer.log_relpath}').upload_from_filename(
+            str(LOGS_DIR / fuzzer.log_relpath))
+
+
+def run_fuzzers(gcs_client: gcs.Client, pause_evt: threading.Event,
+                resume_evt: threading.Event, exit_evt: threading.Event) -> None:
+    """
+    Run all the fuzzers until `exit_evt` gets triggered, pausing and resuming
+    them based on `pause_evt` and `resume_evt`.
+
+    Args:
+        gcs_client: the client to use for syncing with the bucket
+        pause_evt: the pausing event to check
+        resume_evt: the resuming event to check
+        exit_evt: the exiting event to check
+    """
+    # pylint: disable=too-many-locals,too-many-branches,too-many-statements
+
+    bucket = gcs_client.bucket(GCS_BUCKET)
+
+    repo = Repository(REPO_DIR, REPO_URL)
+    repo.clone_if_need_be()
+
+    corpus = Corpus(CORPUS_DIR, bucket)
+
+    while True:
+        if pause_exit_spot(pause_evt, resume_evt, exit_evt):
+            return
+
+        sync_log_files = []
+        date = datetime.datetime.utcnow().strftime('%Y-%m-%d')
+
+        # Read the configuration from the repository
+        cfg = repo.latest_config('master')
+        cfg_for = {
+            b['name']: repo.latest_config(b['name']) for b in cfg['branch']
+        }
+
+        # Figure out which targets we want to run
+        branches = random_weighted(cfg['branch'], NUM_FUZZERS, "branches")
+        targets = [
+            random_weighted(cfg_for[b['name']]['target'], 1, "target")[0]
+            for b in branches
+        ]
```

Comment:
Yup, I removed this code altogether in the latest push, that now also allows for overriding the fuzzing config from master (so that disabling a fuzzer on a release branch doesn't require a backport)